docs/docs/accessibility.md

```diff
@@ -0,0 +1,320 @@
+they include accessibility features.
+
+React fully supports building accessible websites, often by using standard HTML techniques.
+
+## Standards and guidelines:
+
+### WCAG
+
+The [Web Content Accessibility Guidelines](https://www.w3.org/WAI/intro/wcag) provides guidelines for creating accessible web sites.
+
+The following WCAG checklists can help in getting an overview:
+
+- [WCAG checklist from Wuhcag](https://www.wuhcag.com/wcag-checklist/)
+- [WCAG checklist from WebAIM](http://webaim.org/standards/wcag/checklist)
+- [Checklist from The A11Y Project](http://a11yproject.com/checklist.html)
+
+### WAI-ARIA
+
+The [Web Accessibility Initiative - Accessible Rich Internet Applications](https://www.w3.org/WAI/intro/aria) document contains techniques for building fully accessible JavaScript widgets.
+
+Note that all `aria-*` HTML attributes are fully supported in JSX.
+
+## Accessible forms
+
+### Labelling
+Every HTML form contol, such as `<input>` and `<textarea>`, needs to be labelled accessibly. We need to provide descriptive labels that are not only visible but also exposed to screen readers.
+
+The following resources show us how to do this:
+
+- [The W3C shows us how to label elements](https://www.w3.org/WAI/tutorials/forms/labels/)
+- [WebAIM shows us how to label elements](http://webaim.org/techniques/forms/controls)
+- [The Paciello Group explains accessible names](https://www.paciellogroup.com/blog/2017/04/what-is-an-accessible-name/)
+
+Although these standard HTML practices can be directly introduced into React, note that the `for` attribute is written as `htmlFor` in JSX:
+
+```javascript{1}
+<label htmlFor="namedInput">Name:</label>
+<input id="namedInput" type="text" name="name"/>
+```
+
+### Notifying the user of errors
+
+Error situations need to be understood by all users. The following link show us how to expose error texts to screen readers as well:
+
+- [The W3C demonstrates user notifications](https://www.w3.org/WAI/tutorials/forms/notifications/)
+- [WebAIM looks at form validation](http://webaim.org/techniques/formvalidation/)
+
+## Focus control
+
+Ensure that your web application can be fully operated with the keyboard only: 
+
+- [WebAIM talks about keyboard accessibility](https://webaim.org/techniques/keyboard/)
+
+### Keyboard focus and focus outline
+
+Keyboard focus refers to the current element in the DOM that is selected to accept input from the keyboard. We see it everywhere as a focus outline similar to the that shown on the following image:
+
+<center><img src="/react/img/docs/keyboard-focus.png" alt="Blue keyboard focus outline around a selected link." /></center>
+
+Only ever use CSS that removes this outline, for example by setting `outline:0`, if you are replacing it with another focus outline implementation.
+
+### Mechanisms to skip to desired content
+
+Provide a mechanism to allow users to skip past content sections in your application as this assists and speeds up keyboard navigation.
```

Comment:
Yeah. Meant that, as in jump past sections I am not interested in. Fixed in next push :-)